src/west/commands/project.py

```diff
@@ -345,47 +345,16 @@ def do_add_parser(self, parser_adder):
 
     def do_run(self, args, user_args):
         if args.update:
-            _update_west()
+            _update_west(args.rebase, args.keep_descendants)
```

Comment:
There's nothing wrong with this line but GitHub won't let me leave a line comment outside of the hunks in the diff.

The Update description needs to be changed, because -k and -r now affect 1. " West itself is updated to the revision in the manifest file" as well.

I would suggest a new description along these lines:

```
            Updates west and all project repositories according to the
            manifest file, `west.yml`, in the manifest repository.

            By default:

            1. The revisions in the manifest file are fetched from the
            remote in the west repository in the installation and each
            project repository.

            2. The local manifest-rev branches in each repository
            (which are managed exclusively by west) will be hard reset
            to the updated commits fetched in step 1.

            3. All repositories will have detached HEADs checked out
            at the new manifest-rev commits. Locally checked out
            branches are unaffected but will no longer be checked out
            by default.  West will print information on how to get
            back to those branches or update them to the new
            manifest-rev versions.

            You can skip updating west using --no-update.

            You can influence the behavior when local branches are
            checked out using --keep-descendants and/or --rebase.

            This command does not change the contents of the manifest
            repository.
```

Reply:
thanks for the updated text.
Fixed